File: Array/1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        int[] out = {-1, -1};
  
        for(int i=0;i<nums.length;i++) {
            if(map.containsKey(target-nums[i])) {
                out[0] = map.get(target-nums[i]);
                out[1] = i;
            }
            map.put(nums[i], i);
        }
        return out;
    }
}

File: Array/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seenSum = {}

        for i in range(len(nums)):
            num = nums[i]
            if target-num in seenSum:
                return [seenSum[target-num], i]
            seenSum[num] = i
        return [-1,-1]